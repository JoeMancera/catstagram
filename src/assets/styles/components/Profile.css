.head-profile{
  position: absolute;
    margin: auto;
    top: 0;
    right: 0;
    left: 0;
    width: 100vw;
    height: 150px;
    background: var(--primary);
}
.figure-profile {
  position: absolute;
  margin: auto;
  top: 69px;
  right: 0;
  left: 0;
  width: 167px;
  height: 167px;
  border-radius: 50%;
  box-shadow: 0 0 0 5px var(--withe);
  background: var(--secondary);
}

.image-profile {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  object-fit: cover;
}

.profile-user-name{
  position: absolute;
  margin: auto;
  top: -60px;
  right: 0;
  bottom: 0;
  left: 0;
  width: inherit;
  height: 40px;
  text-align: center;
}