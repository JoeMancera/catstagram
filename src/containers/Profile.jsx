import React from 'react';
import catProfileImage from '../assets/static/prof-cat.jpg';
import '../assets/styles/components/Profile.css';

const Profile = ({ user }) => {
  return (
    <>
      <div className='head-profile' />
      <figure className='figure-profile'>
        <img className='image-profile' src={catProfileImage} alt='Cat with tie' />
      </figure>
      <section className='profile-user-name'>
        <h2>Profile</h2>
        <p>{user}</p>
      </section>
    </>
  );
};

export default Profile;
