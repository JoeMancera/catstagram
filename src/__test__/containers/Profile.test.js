import React from 'react';
import { create } from 'react-test-renderer';
import Profile from '../../containers/Profile';
import profileUserMock from '../../__mocks__/ProfileUserMock';

describe('<Profile /> Snapshot test', () => {
  test('Testing UI Componetnt <Profile >', () => {
    const profile = create(<Profile user={profileUserMock} />);

    expect(profile.toJSON()).toMatchSnapshot();
  });
});

