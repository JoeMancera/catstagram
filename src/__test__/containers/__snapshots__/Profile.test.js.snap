// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Snapshot test Testing UI Componetnt <Profile > 1`] = `
Array [
  <div
    className="head-profile"
  />,
  <figure
    className="figure-profile"
  >
    <img
      alt="Cat with tie"
      className="image-profile"
      src="test-file-stub"
    />
  </figure>,
  <section
    className="profile-user-name"
  >
    <h2>
      Profile
    </h2>
    <p>
      catsxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
    </p>
  </section>,
]
`;
